wgpu-core/src/device/any_device.rs

```diff
@@ -1,72 +1,98 @@
+use wgt::Backend;
+
 use super::Device;
 /// The `AnyDevice` type: a pointer to a `Device<A>` for any backend `A`.
 use crate::hal_api::HalApi;
 
-use std::any::Any;
 use std::fmt;
+use std::mem::ManuallyDrop;
+use std::ptr::NonNull;
 use std::sync::Arc;
 
+struct AnyDeviceVtable {
+    // We oppurtunistically store the backend here, since we now it will be used
+    // with backend selection and it can be stored in static memory.
+    backend: Backend,
+    // Drop glue which knows how to drop the stored data.
+    drop: unsafe fn(*mut ()),
+}
+
 /// A pointer to a `Device<A>`, for any backend `A`.
 ///
-/// Any `AnyDevice` is just like an `Arc<Device<A>>`, except that the
-/// `A` type parameter is erased. To access the `Device`, you must
-/// downcast to a particular backend with the \[`downcast_ref`\] or
-/// \[`downcast_clone`\] methods.
-pub struct AnyDevice(Arc<dyn Any + 'static>);
+/// Any `AnyDevice` is just like an `Arc<Device<A>>`, except that the `A` type
+/// parameter is erased. To access the `Device`, you must downcast to a
+/// particular backend with the \[`downcast_ref`\] or \[`downcast_clone`\]
+/// methods.
```

Comment:
nitpick(non-blocking): It's nice to link to referenced methods:

```suggestion
/// particular backend with the [`Self::downcast_ref`] or
/// [`Self::downcast_clone`] methods.
```

Reply:
thought: Side note, your editor seems to soft-break lines to be more narrow than my `neovim` configuration does. 🤔 I wonder why?

Reply:
We probably just have different widths for soft breaking.